src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1422,10 +1426,151 @@ public override IOperation VisitMethodBodyOperation(IMethodBodyOperation operati
         {
             StartVisitingStatement(operation);
 
+            var member = operation.SemanticModel.GetDeclaredSymbol(operation.Syntax);
+            ImmutableArray<IParameterSymbol> parameters = member switch
+            {
+                var x when
+                    x is null ||
+                    x is IFieldSymbol => ImmutableArray<IParameterSymbol>.Empty,
+                IParameterSymbol parameterMember => ImmutableArray.Create(parameterMember),
+                _ => (member as IMethodDefinition).Parameters.CastArray<IParameterSymbol>()
+            };
+
+            VisitNullChecks(operation, parameters, captureIdForResult);
+
             VisitMethodBodyBaseOperation(operation);
             return FinishVisitingStatement(operation);
         }
 
+        internal void VisitNullChecks(IOperation operation, ImmutableArray<IParameterSymbol> parameters, int? captureIdForResult)
+        {
+            var temp = _currentStatement;
+            var nullChecks = GenerateNullChecksForParameters(parameters, operation.Syntax, ((Operation)operation).OwningSemanticModel);
+
+            foreach (var check in nullChecks)
+            {
+                _currentStatement = check;
+                VisitConditional(check, captureIdForResult);
+            }
+            _currentStatement = temp;
+        }
+
+        private ConditionalOperation GenerateNullCheckForParameter(IParameterSymbol parameter, SyntaxNode syntax, SemanticModel semanticModel)
+        {
+            Debug.Assert(parameter.Language == LanguageNames.CSharp);
+            Optional<object> constantValue = default(Optional<object>);
+            var paramReference = new ParameterReferenceOperation(parameter, semanticModel, syntax, parameter.Type, constantValue, isImplicit: true);
+            //IOperation conditionOp = MakeIsNullOperation(paramReference);
```

Comment:
>//IOperation conditionOp = MakeIsNullOperation(paramReference); [](start = 12, length = 63)

Is this commented-out code needed? Should we use `MakeIsNullOperation()` here?

Reply:
It is not needed - I'll remove. Initially I'd tried to replace it with `MakeIsNullOperation()` due to the similarity per @333fred 's suggestion, but I believe it caused an error with the sematic model. 

I wrote down the issue in my notebook, but I don't have it at home. I can try and replace it again and double check.

Reply:
ControlFlowGraphBuilder::6950 gets a null reference exception. Presumably in the owningsemanticmodel. I think it's best to just remove the comment and leave as was.